### ISSUE TITLE
WCAG 2.4.4,WCAG 4.1.2: Ensures links have discernible text (a[data-target="timesheet_edit_form_begin_date"] and others)

### ISSUE THREAD

**Issue by holgerd** [kimai/kimai#4569]
### Describe the issue

I asked my blind colleague, who is a screenreader user, to try out the demo version of Kimai.

She came back to me and said it was unusable for her.

The most pressing issues are the labeling of links and buttons. This can be shown/reproduced with the "Accessibility Insights for Web" plugin by Microsoft (see below). I recommend simply adding labels (I thinkaria labels are not even needed).

**Issue:** Ensures links have discernible text (link-name - https://accessibilityinsights.io/info-examples/web/link-name)

Target application: Kimai – Zeiterfassung - https://demo.kimai.org/de_CH/timesheet/

Element path: a[data-target="timesheet_edit_form_begin_date"]

Snippet: <a href="#" data-form-widget="date-now" data-format="DD.MM.YYYY" data-target="timesheet_edit_form_begin_date"><i class="far fa-calendar-alt"></i></a>

How to fix: 
Fix all of the following:
  Element is in tab order and does not have accessible text
  https://accessibilityinsights.io/info-examples/web/link-name/

Fix any of the following:
  Element does not have text that is visible to screen readers
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element has no title attribute

Environment: Chrome version 120.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.41.0 (axe-core 4.7.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

probably all of them

### How do you run Kimai?

KIMAI CLOUD

### Which PHP version are you using?

Unknown

**Comment by holgerd** [comment]
There are 57 instances of this issue according to the automated tool. There might be more in other pages, I just checked one.

**Comment by kevinpapst** [comment]
Thank you for raising this topic. 

I will work on these issues one after one, but I need real user feedback. 
These tools provide basic information, which are (more or less easy) solvable. 
But I guess that the more problematic issues are hidden deeper. 

Maybe your colleague can re-test after the next update?
And then it would be great if there was more feedback than just "it is unusable". 
Which is good to know, but not really helping me to improve the situation.

And I will ONLY work on the parts for now, which are user focussed. 
I am not going into the rabbit hole of the admin areas yet.

**Comment by holgerd** [comment]
Sounds great. Yes, she and I are happy to test and provide more feedback.

**Comment by kevinpapst** [comment]
Can you please test the "Create timesheet" on a single screen with her?
Simply right click the <img width="166" alt="Bildschirmfoto 2024-01-17 um 17 29 12" src="https://github.com/kimai/kimai/assets/533162/76781d2d-fa2e-468b-bc4e-517b13f4661e"> button and then "open in new tab"?

I am especially interested in the form. 
What about the date and time input and especially what about the §enhanced dropdowns" (e.g. Customer and Project). Those are javascript libraries where I only have little control over. I'd like to understand whether those are usable or if we have to find a way to switch back to native browser elements (which are likely more optimized for accessibility).

**Comment by holgerd** [comment]
Thanks for working on this., She told me she could fill out the form now. :) 

The most pressing issue for her is that the buttons still have no labels, making it impossible for her to discern them from each other. So the form fields work, but she still cannot record the tracked time.

**Comment by kevinpapst** [comment]
Thanks 👍 

As I said: one step after another. Happy to improve that, but it is hard for me, because I do not understand what is not working. Which buttons don't have labels? 

Afaik if the button has a text, then it will be read, if there is no additional aria attribute?!
Save and cancel are labeled. Or is she talking about these enhanced form field stuff like "set date to now" or "open begin dropdown to select a time"?

Or do you mean "input fields"? If I look at the form, I think the first two rows only have one label but two fields....

If you have more input or recommendations, let me know, otherwise I'll read more about the topic and try to find improvements by myself.

Edit: partially explained in the title of the issue 😁
